src/nlp_types.jl

```diff
@@ -212,3 +215,59 @@ function print(io :: IO, nlp :: NLPModelMeta)
     @printf(io, "nonlinear network constraints:   "); dsp(nlp.nnet'); @printf(io, "\n");
   end
 end
+
+"""
+    has_bounds(nlp)
+    has_bounds(meta)
+
+Returns whether the problem has bounds on the variables.
+"""
+has_bounds(meta::NLPModelMeta) = length(meta.ifree) < meta.nvar
+has_bounds(nlp::AbstractNLPModel) = has_bounds(nlp.meta)
```

Comment:
If you wanted, and in case we add more such methods, those could all be defined at once:
```julia
for meth in (:has_bounds, :bound_constrained, ...)
  @eval $meth(nlp::AbstractNLPModel) = $meth(nlp.meta)
end
```